### PR TITLE
Add mentors empty state page for claim flow

### PR DESCRIPTION
**Pull request by Nitemaeric**
## Context

When a User attempts to select mentors for a claim that have all fully been claimed for with the provider selected, we should show a message that stops users from proceeding further.

## Changes proposed in this pull request

- Add a variant of the mentor select page that shows an error message when 
- Update `spec/system/claims/support/schools/claims/create_claim_spec.rb` to only create a single school. This removes the flakiness of this spec suite when it randomly creates 2 schools with the same name.

## Guidance to review

- Fully claim all hours for a school's mentors against a provider.
- Try to claim for a mentor against that provider.
- It should say that you have no mentors you can claim for.

## Screenshots

CleanShot 2024-05-01 at 15 25 15

https://github.com/DFE-Digital/itt-mentor-services/assets/4231530/13de0a64-6395-4472-a110-44646b26aea2

### CODE REVIEW

File: app/controllers/claims/schools/claims/mentors_controller.rb [hunk 1]
```diff
@@ -8,13 +8,17 @@ def new; end
...
   def create
     if claim_mentors_form.save
-      redirect_to(
-        edit_claims_school_claim_mentor_training_path(
-          @school,
-          claim,
-          claim.mentor_trainings.without_hours.first,
-        ),
-      )
+      if claim.mentor_trainings.without_hours.any?
```
Comment (CatalinVoineag) [inline app/controllers/claims/schools/claims/mentors_controller.rb:11]: This probably should be a method in the form object but considering the situation is fine.

File: app/views/claims/schools/claims/check.html.erb [hunk 2]
```diff
@@ -62,6 +64,7 @@
                                  @claim,
                                  mentor_training,
                                ),
+                               visually_hidden_text: "Hours of training for #{mentor_training.mentor.full_name}",
```
Comment (CatalinVoineag) [inline app/views/claims/schools/claims/check.html.erb:67]: This should probably be a translation but I don't mind

**Review by CatalinVoineag — APPROVED** [review]
Played around and it looks good to me 👍🏻

File: app/controllers/claims/schools/claims/mentors_controller.rb (hunk 1 above)
Comment (Nitemaeric) [reply app/controllers/claims/schools/claims/mentors_controller.rb:11]: Agreed. I'll clean this up as soon as I can. Thanks!

**Pull request merged**